lib/imap.c

```diff
@@ -1025,6 +1026,9 @@ static CURLcode imap_state_auth_resp(struct Curl_easy *data,
     default:
       break;
     }
+  else if((!imapc->login_disabled) && (imapc->preftype & IMAP_TYPE_CLEARTEXT))
```

Comment:
Ref: https://datatracker.ietf.org/doc/html/rfc9051#section-6.2.3

Though the burden is on the server to provide logindisabled should this fallback be allowed without ssl, in other words why not this:

```suggestion
  else if(!imapc->login_disabled && imapc->ssldone &&
          (imapc->preftype & IMAP_TYPE_CLEARTEXT))
```

Reply:
While I support the prohibition of plain-text passwords without TLS, I'm a bit afraid that this would break scripts.

Reply:
I think the login disabled flag is sufficient as we do for the other use of cleartext in IMAP.

Not only that but introducing the dependency on TLS would prevent curl or libcurl being used in systems where other mechanisms are in place to secure the network - such as a closed wall system that has no outside access.

Reply:
I note that this change has broken test849 - whilst I have fixed it up locally to add the fallback I am still deliberating whether cleartext fallback should be used or not when an authzid has been provided.

Does anyone have any strong views either way?